src/overlays/actors/ovl_En_Sb/z_en_sb.c

```diff
@@ -21,44 +36,373 @@ const ActorInit En_Sb_InitVars = {
+    if (this->actor.velocity.y <= -0.1f || this->actor.bgCheckFlags & 2) {
+        if (!(this->actor.yDistToWater > 0.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_EYEGOLE_ATTACK);
+        }
+        this->actor.bgCheckFlags &= ~2;
+        EnSb_SetupBounce(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/func_808E4458.asm")
+void EnSb_Bounce(EnSb* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 currentFrame;
+    f32 frameCount;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/func_808E44D4.asm")
+    currentFrame = this->skelAnime.animCurrentFrame;
+    frameCount = SkelAnime_GetFrameCount(&D_060000B4.common);
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 0.2f);
+    if (currentFrame == frameCount) {
+        if (this->bounceCounter != 0) {
+            this->bounceCounter--;
+            this->attackTimer = 1;
+            if (this->actor.yDistToWater > 0.0f) {
+                this->actor.velocity.y = 3.0f;
+                this->actor.speedXZ = 5.0f;
+                this->actor.gravity = -0.35f;
+            } else {
+                this->actor.velocity.y = 2.0f;
+                this->actor.speedXZ = 6.0f;
+                this->actor.gravity = -2.0f;
+            }
+            EnSb_SpawnBubbles(globalCtx, this);
+            EnSb_SetupLunge(this);
+        } else if (this->actor.bgCheckFlags & 1) {
+            this->actor.bgCheckFlags &= ~2;
+            this->actor.speedXZ = 0.0f;
+            this->attackTimer = 1;
+            EnSb_SetupWaitClosed(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/func_808E45A0.asm")
+void EnSb_ReturnToIdle(EnSb* this, GlobalContext* globalCtx) {
+    if (this->attackTimer != 0) {
+        this->attackTimer--;
+        if (this->actor.bgCheckFlags & 1) {
+            this->actor.bgCheckFlags &= ~1;
+            this->actor.speedXZ = 0.0f;
+        }
+    } else if (this->actor.bgCheckFlags & 1) {
+        this->actor.bgCheckFlags &= ~1;
+        this->actionFunc = EnSb_Idle;
+        this->actor.speedXZ = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/func_808E4654.asm")
+void EnSb_UpdateDamage(EnSb* this, GlobalContext* globalCtx) {
+    Vec3f hitPoint;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/func_808E4740.asm")
+    if (this->collider.base.acFlags & 2) {
+        s32 hitPlayer = 0;
+        this->collider.base.acFlags &= ~2;
+        if (this->actor.colChkInfo.damageEffect == 0xF) {
+            hitPlayer = 0;
+            if (this->vulnerableTimer != 0) {
+                func_800BE22C(&this->actor);
+                func_800BCB70(&this->actor, 0x4000, 0xFF, 0x2000, 80);
+                hitPlayer = 1;
+            }
+        }
+        if (hitPlayer != 0) {
+            this->unk_252 = 0;
+            if (this->actor.draw != 0 && this->unk_253 == 0) {
```

Comment:
```suggestion
            if ((this->actor.draw != NULL) && (this->unk_253 == 0)) {
```
`this->actor.draw` takes function pointers iirc, so `NULL` is appropriate